src/pudl/analysis/ferc1_eia.py

```diff
@@ -0,0 +1,1533 @@
+            # Train the classifier
+            lrc.fit(x_train, y_train)
+            # predict matches for the test
+            result_lrc = lrc.predict(x_test)
+            # generate and compile the scores and outcomes of the
+            # prediction
+            fscore.append(rl.fscore(y_test, links_pred=result_lrc))
+            precision.append(rl.precision(y_test, links_pred=result_lrc))
+            accuracy.append(
+                rl.accuracy(y_test, links_pred=result_lrc, total=result_lrc)
+            )
+            result_lrc_complied = pd.concat(
+                [result_lrc_complied, pd.DataFrame(index=result_lrc)]
+            )
+        return result_lrc_complied, fscore, precision, accuracy
+
+    def fit_predict_option(
+        self, solver, c, cw, p, l1, n_splits, multi_class, results_options
+    ):
+        """Test and cross validate with a set of model parameters.
+
+        In this method, we instantiate a model object with a given set of
+        hyperparameters (which are selected within `test_model_parameters`)
+        and then run k-fold cross vaidation with that model and our training
+        data.
+
+        Returns:
+            pandas.DataFrame
+        """
+        logger.debug(f"train: {solver}: c-{c}, cw-{cw}, p-{p}, l1-{l1}")
+        lrc = rl.LogisticRegressionClassifier(
+            solver=solver,
+            C=c,
+            class_weight=cw,
+            penalty=p,
+            l1_ratio=l1,
+            random_state=0,
+            multi_class=multi_class,
+        )
+        results, fscore, precision, accuracy = self.kfold_cross_val(
+            lrc=lrc,
+            n_splits=n_splits,
+            features_known=self.features_train,
+            known_index=self.train_index,
+        )
+
+        # we're going to want to choose the best model so we need to save the
+        # results of this model run...
+        results_options = pd.concat(
+            [
+                results_options,
+                pd.DataFrame(
+                    data={
+                        # result scores
+                        "precision": [statistics.mean(precision)],
+                        "f_score": [statistics.mean(fscore)],
+                        "accuracy": [statistics.mean(accuracy)],
+                        # info about results
+                        "coef": [lrc.coefficients],
+                        "interc": [lrc.intercept],
+                        "predictions": [len(results)],
+                        # info about which model hyperparameters we choose
+                        "solver": [solver],
+                        "c": [c],
+                        "cw": [cw],
+                        "penalty": [p],
+                        "l1": [l1],
+                        "multi_class": [multi_class],
+                    },
+                ),
+            ]
+        )
+        return results_options
+
+    @staticmethod
+    def get_hyperparameters_options():
```

Comment:
We could pretty easily generate a static little dict of just *one* set of hyperparameter options. Add a param here that could either take the default hyperparams or generate them all. This way in `test_model_parameters`, we'd iterate over just one option.

Very open to other options here... I set this up knowing very litttttle about how to properly test hyperparams/what they actually impact and still feel pretty ignorant.

These seem to be the consistent options:
<img width="340" alt="image" src="https://user-images.githubusercontent.com/25487998/214155179-468414bc-b6ba-49d7-a451-9d7f44bea926.png">

Reply:
To start with, I think we want to leave it in the hands of `sklearn` as much as possible. For our `rl.LogisticRegressionClassifier` model (the kernel of which is just `sklearn.linear_model.LogisticRegression`) I think exhaustive Grid Search ([sklearn.model_selection.GridSearchCV](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GridSearchCV.html#sklearn.model_selection.GridSearchCV)) is pretty standard. Grid Search is going to iterate through all the potential parameters (like you did below with looping) but with a matrix of parameters instead. I think it should be faster.

So you can do something like:
```
clf = GridSearchCV(estimator=lrc, param_grid = param_grid, cv = 5, verbose=True, n_jobs=-1)
best_clf = clf.fit(X_train, y_train)
```
cv = 5 is the default 5 fold cross validation which I think you are using here.

Will this speed things up enough that we can leave in the hyperparameter tuning? 

That being said, my brain maybe wasn't processing this hyperparameter tuning process correctly, so forgive me if I'm missing something here.

[Potentially helpful logistic regression + grid search example](https://towardsdatascience.com/logistic-regression-model-tuning-with-scikit-learn-part-1-425142e01af5)
[Potentially helpful general scikitlearn doc on tuning parameters](https://scikit-learn.org/stable/modules/grid_search.html)

Reply:
P.S. It seems like `sklearn` has an estimator called [LogisticRegressionCV](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegressionCV.html#sklearn.linear_model.LogisticRegressionCV) that has built in cross-validation capabilities that speed things up. Basically I think it will warm start each model run on a new set of parameters with values from a previous run. Anyways. It would be nice if we could use this `LogisticRegressionCV` instead of `LogisticRegression` estimator but it seems like the `recordlinkage` library only has `LogisticRegression` as a [classifier kernel option](https://recordlinkage.readthedocs.io/en/latest/ref-classifiers.html). Do we need to use a built in `recordlinkage` library classifier or could we potentially use `sklearn.linear_model.LogisticRegressionCV`? This whole comment might be too much in the weeds. Just throwing it out there.